### ISSUE TITLE
Add basic passive checks

### ISSUE THREAD

**Issue by sw33tLie** [caido/caido#100]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help us know what to work on
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Description

One feature I really miss in Caido is a basic reflection detector.
On Burp Suite, I rely on [this extension](https://github.com/elkokc/reflector) and I've often found interesting bugs even by mistake just by using it.

I am not sure about the devs' plans on the active scanner, but a few passive checks would be awesome to have and could greatly improve Caido's usage experience.
I believe most of them are not even time-consuming to code as it's usually just a matter of matching strings.

A few checks that I'd like to have are:
- Reflection detector (for XSS & HTML injection)
- Source code exposure (e.g. `<?php` somewhere in a webpage that was proxied)
- Software versions disclosed (e.g. Tomcat xxx, nginx yyy)
- Credit card numbers disclosed
- Disclosed emails
- Disclosed IP addresses
- Suspicious error messages
- More?

Let me know your thoughts on this.
Thanks!

**Comment by Sytten** [comment]
I think that would go nicely in a plugin? So the community doesn't have to rely on us to add new checks.
We might do "native" plugins that you can replace by more advanced ones.

**Comment by Corb3nik** [comment]
This feature might actually be a good candidate to open source rather than as plugins. 

We could have a Github repository containing nothing but scanner checks, which would then be compiled into our release builds.

**Comment by Corb3nik** [comment]
Basically Nuclei but for passive checks: https://github.com/projectdiscovery/nuclei-templates

**Comment by sw33tLie** [comment]
Yep, I was wondering if you were seeing this fit more like a plugin or directly built-in.
Whatever works, of course.
Maybe if you choose the plugin route I'd make it installed by default as it's just too useful to have :)

**Comment by Corb3nik** [comment]
Considered this done via https://github.com/caido-community/scanner
